tools/cldr-apps/src/main/java/org/unicode/cldr/web/STFactory.java

```diff
@@ -230,11 +231,41 @@ private VoteResolver<String> setValueFromResolver(String path, VoteResolver<Stri
             }
             delegate.removeValueAtDPath(path);
             if (value != null) {
+                value = reviseInheritanceAsNeeded(path, value);
                 delegate.putValueAtPath(fullPath, value);
             }
             return resolver;
         }
 
+        /**
+         * Get the possibly modified value. If value matches the bailey value or inheritance marker, possibly
+         * change it from bailey value to inheritance marker, or vice-versa, as needed to meet these requirements:
+         * 1. If the path changes when getting bailey, then we are inheriting sideways. We need to use a hard value.
+         * 2. If the value is different from the bailey value, can't use inheritance; we need a hard value.
+         * 3. Otherwise we use inheritance marker.
+         *
+         * @param path the path
+         * @param value the input value
+         * @return the possibly modified value
+         */
+        private String reviseInheritanceAsNeeded(String path, String value) {
+            if (value != null) {
+                CLDRFile cldrFile = make(ballotBox.locale.getBaseName());
+                if (cldrFile != null) {
+                    if (!cldrFile.isResolved()) {
+                        throw new InternalCldrException("must be resolved");
+                    }
+                    Output<String> pathWhereFound = new Output<>();
+                    Output<String> localeWhereFound = new Output<>();
+                    String baileyValue = cldrFile.getBaileyValue(path, pathWhereFound, localeWhereFound);
+                    if (baileyValue != null && (CldrUtility.INHERITANCE_MARKER.equals(value) || baileyValue.equals(value))) {
+                        return pathWhereFound.value.equals(path) ? CldrUtility.INHERITANCE_MARKER : baileyValue;
+                    }
+                }
+             }
+            return value;
+        }
```

Comment:
I don't think this belongs here.  It should go as a static function on CLDRFile so it can be unit tested separately.

Reply:
good point